### PR TITLE
Document support for custom pip index servers

### PR DESCRIPTION
**Pull request by chmeliik**
[STONEBLD-2506](https://issues.redhat.com/browse/STONEBLD-2506)

### CODE REVIEW

**Review by arewm — APPROVED** [review]

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 1]
```diff
@@ -204,6 +204,60 @@ spec:
 * From the {ProductName} *Applications* view, go to *Activity > Latest commits*.
 //include::../partials/con_hermetic_verification.adoc[]
 
+=== Prefetching `pip` dependencies from custom index servers
+
+Cachi2 supports the link:https://pip.pypa.io/en/stable/cli/pip_install/#install-index-url[--index-url] option.
+You can add this option to your `requirements.txt` file(s), instructing Cachi2 to download packages from the specified
+index server. Your file may look something like this:
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:211]:
```suggestion
index server. For example:
```

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 2]
```diff
@@ -204,6 +204,60 @@ spec:
 * From the {ProductName} *Applications* view, go to *Activity > Latest commits*.
 //include::../partials/con_hermetic_verification.adoc[]
 
+=== Prefetching `pip` dependencies from custom index servers
+
+Cachi2 supports the link:https://pip.pypa.io/en/stable/cli/pip_install/#install-index-url[--index-url] option.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:209]: Is this correct? `--index-url`

or do we need to use:

```suggestion
Cachi2 supports the link:https://pip.pypa.io/en/stable/cli/pip_install/#install-index-url[index url] option.
```

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 3]
```diff
@@ -204,6 +204,60 @@ spec:
...
+----
+--index-url=https://example.pypi.org/simple/
+requests==2.32.2 \
+    --hash=sha256:dd951ff5ecf3e3b3aa26b40703ba77495dab41da839ae72ef3c8e5d8e2433289 \
+    --hash=sha256:fc06670dd0ed212426dfeb94fc1b983d917c4f9847c863f313c9dfaaffb7c23c
+
+# ...other packages
+----
+
+WARNING: *Do not include credentials in the index url.* If needed, provide authentication via a `.netrc` file (as described below).
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:223]:
```suggestion
WARNING: Do not include credentials in the index url. If needed, provide authentication through `.netrc` file (as described below).
```

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 4]
```diff
@@ -204,6 +204,60 @@ spec:
...
+
+To provide `.netrc` authentication for the index server:
+
+. Create a key/value secret (see xref:/how-tos/configuring/creating-secrets.adoc[creating secrets for your builds]).
+  Set `.netrc` as the key. Upload the content of your `.netrc` file as the value. For more details on netrc files,
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:228]: Upload where?

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 5]
```diff
@@ -204,6 +204,60 @@ spec:
...
+  see https://pip.pypa.io/en/stable/topics/authentication/#netrc-support.
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:229]:
```suggestion
  see link:https://pip.pypa.io/en/stable/topics/authentication/#netrc-support[netrc support].
```

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc (hunk 4 above)
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:228]:
```suggestion
  . Set `.netrc` as the key. 
  . Upload the content of your `.netrc` file as the value. For more infromation on netrc files,
```

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 6]
```diff
@@ -204,6 +204,60 @@ spec:
...
+
+. In your `.tekton/` PipelineRun files, pass the netrc secret to the pipeline. Add the newly created secret to the `.spec.workspaces` section:
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:231]:
```suggestion
. In your `.tekton/` PipelineRun files, pass the netrc secret to the pipeline. 
. Add the newly created secret to the `.spec.workspaces` section:
```

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc [hunk 7]
```diff
@@ -204,6 +204,60 @@ spec:
...
+spec:
+  # ...
+  workspaces:
+    # ...
+    - name: netrc
+      secret:
+        secretName: my-netrc  # the name you assigned to the secret in step 1
+----
+
+. Make sure your PipelineRuns pass the netrc workspace to the prefetch-dependencies task. In the `.spec.pipelineSpec.tasks` section,
```
Comment (gtrivedi88) [inline docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:245]:
```suggestion
. Make sure your PipelineRuns pass the netrc workspace to the prefetch-dependencies task. 

. In the `.spec.pipelineSpec.tasks` section,
```

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc (hunk 2 above)
Comment (chmeliik) [reply docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:209]: `--index-url` is how the pip documentation refers to it and how it appears in the requirements.txt file

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc (hunk 4 above)
Comment (chmeliik) [reply docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:228]: image

I think this should be covered by the linked creating-secrets doc

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc (hunk 6 above)
Comment (chmeliik) [reply docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc:231]: It is one step - the user passes the netrc secret to the pipeline by adding it to `.spec.workspaces`

Comment (chmeliik) [reply docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc]: I dropped `pass the netrc secret to the pipeline.`. It was meant to explain what the user achieves by adding the workspace, but I think it only added confusion

File: docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc (hunk 7 above)
Comment (chmeliik) [reply docs/modules/ROOT/pages/how-tos/configuring/prefetching-dependencies.adoc]: Same here, dropped the whole first sentence

**Comment by chmeliik** [comment]
Rebased on main. FYI @gtrivedi88 I think I've addressed your comments, PTAL

**Review by gtrivedi88 — APPROVED** [review]

**Pull request merged**